source/connection.c

```diff
@@ -963,6 +1051,27 @@ int aws_http_client_connect_internal(
         http_bootstrap->http2_options.initial_settings_array = setting_array;
     }
 
+    if (options.alpn_string_map) {
+        alpn_string_map = aws_http_default_alpn_map_new(options.allocator);
```

Comment:
I vote that we be explicit when the user passes in an ALPN map: Copy it exactly, and reject anything that's not in the list.

It's a little weird to automatically insert "h2" and "http/1.1" when the user didn't ask. It's also a little weird to fall back to HTTP/1.1 if the negotiated string doesn't match anything. We added the fallback to HTTP/1.1 when we needed to support a custom ALPN string, but had no controls for doing so. But now we DO have a control for doing so.

Reply:
oh. I was confused. you ARE copying it exactly.

I was thrown off by the word "default" in `aws_http_default_alpn_map_new()`
How about just `aws_http_alpn_map_new()`